packages/SwingSet/test/promise-watcher/test-promise-watcher.js

```diff
@@ -0,0 +1,115 @@
+    },
+    bundles: {
+      upton: { sourceSpec: bfile('vat-upton.js') },
+    },
+  };
+
+  const unhandledRejections = [];
+  handleUnhandledRejections(rej => unhandledRejections.push(rej));
+
+  const hostStorage = provideHostStorage();
+  // ? const { kvStore } = hostStorage;
+  await initializeSwingset(config, [], hostStorage);
+  const c = await makeSwingsetController(hostStorage);
+  c.pinVatRoot('bootstrap');
+  await c.run();
+
+  async function run(name, args = []) {
+    assert(Array.isArray(args));
+    const kpid = c.queueToVatRoot('bootstrap', name, capargs(args));
+    await c.run();
+    const status = c.kpStatus(kpid);
+    const capdata = c.kpResolution(kpid);
+    return [status, capdata];
+  }
+
+  // create initial version
+  const [v1status /* , v1capdata */] = await run('buildV1', []);
+  t.is(v1status, 'fulfilled');
+  const beforeReference = [
+    'lp1-pw resolved lval1 version v1 via watcher []',
+    'lp1-dk resolved lval1 version v1 via VDO',
+    'lp2-pw rejected lerr2 version v1 via watcher []',
+    'lp2-dk rejected lerr2 version v1 via VDO',
+    'rp1-pw resolved rvalbefore version v1 via watcher []',
+    'rp1-dk resolved rvalbefore version v1 via VDO',
+    'rp2-pw rejected rerrbefore version v1 via watcher []',
+    'rp2-dk rejected rerrbefore version v1 via VDO',
+    'p1-pw resolved val1 version v1 via watcher []',
+    'p1-dk.full resolved val1 version v1 via VDO',
+    'p1-dk.res resolved val1 version v1 via VDO (res)',
+    'p2-pw rejected err2 version v1 via watcher [a]',
+    'p2-dk.full rejected err2 version v1 via VDO',
+    'p2-dk.rej rejected err2 version v1 via VDO (rej)',
+  ];
+  t.deepEqual(c.dump().log, beforeReference);
+  if (os.platform() === 'darwin') {
+    // this will not work in CI but can at least validate things locally
+    t.deepEqual(unhandledRejections, ['err2']);
+  }
+
+  // now perform the upgrade
+  // console.log(`-- starting upgradeV2`);
+  const [v2status /* , v2capdata */] = await run('upgradeV2', []);
+  t.is(v2status, 'fulfilled');
+  t.deepEqual(c.dump().log, [
+    ...beforeReference,
+    'lp3-pw rejected vat upgraded version v2 via watcher []',
+    'lp3-dk rejected vat upgraded version v2 via VDO',
+    'lp4-pw rejected vat upgraded version v2 via watcher []',
+    'lp4-dk rejected vat upgraded version v2 via VDO',
+    'p3-pw resolved val3 version v2 via watcher [b,c]',
+    'p3-dk.full resolved val3 version v2 via VDO',
+    'p3-dk.res resolved val3 version v2 via VDO (res)',
+    'p4-pw rejected err4 version v2 via watcher [d,e,f]',
+    'p4-dk.full rejected err4 version v2 via VDO',
+    'p4-dk.rej rejected err4 version v2 via VDO (rej)',
+    'rp3-pw resolved rvalafter version v2 via watcher []',
+    'rp3-dk resolved rvalafter version v2 via VDO',
+    'rp4-pw rejected rerrafter version v2 via watcher []',
+    'rp4-dk rejected rerrafter version v2 via VDO',
+  ]);
+  if (os.platform() === 'darwin') {
```

Comment:
FWIW, I was able to remove this check on my pseudo-linux box (macOS running a linux kernel under Docker using Node v16.14.0) and also a real linux box (also with Node v16.14.0). What version of Node are you running under macOS locally?

I'm not concerned deeply about testing the unhandled rejection thing, but the `os.platform` check is sorta awkward, so after this lands, let's see if we can get rid of it. Maybe we just need to be running a newer version of Node in CI.